Tutorial: How to set up a location based site banner (#6128)
PostHog/posthog.com: contents/tutorials/location-based-banner.md

---
title: How to set up a location-based site banner
date: 2023-06-14
author: ["ian-vanagas"]
showTitle: true
sidebar: Docs
featuredImage: ../images/tutorials/banners/tutorial-3.png
tags: ['feature flags']
---

> Many sites want to set up banners to display information for different users, such as regional announcements or country-based alerts. Doing this with feature flags is simple and this tutorial shows you how to set it up.

Reviewer: hyperlink "feature flags" to docs

> ## Create a Next.js app

Reviewer: I would add a line/paragraph to the intro saying something like "in this tutorial we do this with Next JS, but this is applicable to any framework"

> The first step is creating a basic app, which we Next.js to do. Run the script below, chose no to using TypeScript, yes to using the app router, and the defaults for everything else.

Reviewer: ```suggestion
The first step is creating a basic app, which we Next.js to do. Run the script below, choose no to using TypeScript, yes to using the app router, and the defaults for everything else.
```

```js
// app/providers.js
'use client'
import posthog from 'posthog-js'
import { PostHogProvider } from 'posthog-js/react'

if (typeof window !== 'undefined') {
  posthog.init(
    '<ph_project_api_key>',{
      api_host:'<ph_instance_address>'
    }
  )
}

export default function PHProvider({ children }) {
  return <PostHogProvider client={posthog}>{children}</PostHogProvider>
}

```

Once we created the provider, we can add it to our `layout.js` file.

```js
// app/layout.js
import './globals.css'
import Providers from './providers'

export default function RootLayout({ children }) {
  return (
    <html lang="en">
      <Providers>
        <body>{children}</body>
      </Providers>
    </html>
  )
}
```

Finally, we can run the app with `npm run dev`. When we go to the page running locally, we see events captured into our PostHog instance.

![Events](../images/tutorials/location-based-banner/events.png)

## Set up the banner feature flag

> First, make sure to activate the [GeoIP app](https://app.posthog.com/project/apps?name=geoip) (it should be by default).

Reviewer: ```suggestion
First, make sure that the [GeoIP app](https://app.posthog.com/project/apps?name=geoip) is enabled (it should be by default).
```

> Once you confirm that, we can create our feature flag controlling the site banner. Go to feature flags, click "Create new flag," and set the key as "site-banner." For release conditions, match users where the continent code equals NA (or whatever geography you want), roll out to 100% of users, and press save.

Reviewer: ```suggestion
Once you have confirmed that, we can create our feature flag controlling the site banner. Go to [feature flags](https://app.posthog.com/feature_flags), click "Create new flag," and name the key as "site-banner." For release conditions, match users where the continent code equals NA (or whichever geography you want), roll out to 100% of users, and press save.
```
